src/systems/model_parsing.jl

```diff
@@ -282,9 +282,20 @@ function generate_var(a, varclass;
     var
 end
 
+singular(sym) = last(string(sym)) == 's' ? Symbol(string(sym)[1:(end - 1)]) : sym
+
+function check_name_uniqueness(dict, a, newvarclass)
+    for varclass in [:variables, :parameters, :structural_parameters, :constants]
+        if haskey(dict, varclass) && a in keys(dict[varclass])
```

Comment:
```suggestion
        dvarclass = get(dict, varclass, nothing)
        if dvarclass !== nothing && a in keys(dvarclass)
```
since this function will be called on all systems, it might be nice to have it be performant. This change reduces the number of hash lookups in the if condition from 3 to 2